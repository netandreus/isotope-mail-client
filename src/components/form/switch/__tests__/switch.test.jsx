import React from 'react';
import {shallow} from 'enzyme/build/index';
import Switch from '../switch';

describe('Checkbox component test suite', () => {
  test('Snapshot render, should render switch', () => {
    const {id, label, required, checked, onToggle, switchClass, inputClass} = {
      id: 'Bourne', label: 'White label',
      required: true, checked: true, onToggle: () => {},
      switchClass: 'switch-class', inputClass: 'input-class'};
    const switchh = shallow(<Switch id={id} label={label} required={required} checked={checked}
      onToggle={onToggle} switchClass={switchClass} inputClass={inputClass}/>);
    expect(switchh).toMatchSnapshot();
  });
  test('click, should trigger onToggle function', () => {
    // Given
    const onToggle = jest.fn();
    const preventDefault = jest.fn();
    const switchh = shallow(<Switch onToggle={onToggle}/>);

    // When
    switchh.simulate('click', {preventDefault: preventDefault});

    // Then
    expect(preventDefault).toHaveBeenCalledTimes(1);
    expect(onToggle).toHaveBeenCalledTimes(1);
  });
});
