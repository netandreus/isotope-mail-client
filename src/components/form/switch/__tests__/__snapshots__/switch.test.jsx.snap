// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component test suite Snapshot render, should render switch 1`] = `
<span
  onClick={[Function]}
>
  <div
    className="mdc-switch mdc-switch--checked switch-class"
  >
    <div
      className="mdc-switch__track"
    />
    <div
      className="mdc-switch__thumb-underlay"
    >
      <div
        className="mdc-switch__thumb"
      >
        <input
          checked={true}
          className="mdc-switch__native-control input-class"
          id="Bourne"
          readOnly={true}
          required={true}
          role="switch"
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <label
    htmlFor="Bourne"
  >
    White label
  </label>
</span>
`;
